src/alfasim_score/converter/alfacase/score_input_reader.py

```diff
@@ -17,3 +28,65 @@ def read_well_trajectory(self) -> Tuple[Array, Array]:
+        tubing_data = []
+        for section in self.input_content["operation"]["tubing_string"]["string_sections"]:
+            thermal_property = section["pipe"]["grade"]["thermomechanical_property"]
+            tubing_data.append(
+                {
+                    "name": section["pipe"]["grade"]["name"],
+                    "density": Scalar(thermal_property["density"], DENSITY_UNIT),
+                    "thermal_conductivity": Scalar(
+                        thermal_property["thermal_conductivity"], THERMAL_CONDUCTIVITY_UNIT
+                    ),
+                    "specific_heat": Scalar(thermal_property["specific_heat"], SPECIFIT_HEAT_UNIT),
+                    "thermal_expansion": Scalar(
+                        thermal_property["thermal_expansion_coefficient"], THERMAL_EXPANSION_UNIT
+                    ),
+                    "young_modulus": Scalar(thermal_property["e"], YOUNG_MODULUS_UNIT),
+                    "poisson_coefficient": Scalar(thermal_property["nu"], FRACTION_UNIT),
+                }
+            )
+        return tubing_data
+
+    def read_cement_material(self) -> Dict[str, Union[Scalar, str]]:
+        """ "Read the data for the cement from SCORE input file"""
+        properties = self.input_content["well_strings"][0]["cementing"]["first_slurry"][
+            "thermomechanical_property"
+        ]
+        return {
+            "name": "cement",
+            "density": Scalar(properties["density"], DENSITY_UNIT),
+            "thermal_conductivity": Scalar(
+                properties["thermal_conductivity"], THERMAL_CONDUCTIVITY_UNIT
+            ),
+            "specific_heat": Scalar(properties["specific_heat"], SPECIFIT_HEAT_UNIT),
+            "thermal_expansion": Scalar(
+                properties["thermal_expansion_coefficient"], THERMAL_EXPANSION_UNIT
+            ),
+            "young_modulus": Scalar(properties["e"], YOUNG_MODULUS_UNIT),
+            "poisson_coefficient": Scalar(properties["nu"], FRACTION_UNIT),
+        }
+
+    def read_lithology_materials(self) -> List[Dict[str, Union[Scalar, str]]]:
+        """ "Read the data for the lithologies from SCORE input file"""
+        lithology_data = []
+        for lithology in self.input_content["lithologies"]:
+            properties = lithology["thermomechanical_property"]
+            lithology_data.append(
+                {
+                    "name": lithology["display_name"],
+                    "density": Scalar(properties["density"], DENSITY_UNIT),
+                    "thermal_conductivity": Scalar(
+                        properties["thermal_conductivity"], THERMAL_CONDUCTIVITY_UNIT
+                    ),
+                    "specific_heat": Scalar(properties["specific_heat"], SPECIFIT_HEAT_UNIT),
+                    # expansion has null value, so just use alfacase default 0.0 for this parameter
+                    "thermal_expansion": Scalar(0.0, THERMAL_EXPANSION_UNIT),
```

Comment:
Good. In APB model it is assumed ZERO indeed. You can improve comment considering it.

Reply:
Done